sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where(
+        new Brackets((qb) =>
+          qb
+            .where("applicationOfferingChangeRequest.id IS NULL")
+            .orWhere(
+              "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus NOT IN (:...status)",
+              {
+                status: [
+                  ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+                  ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+                ],
+              },
+            ),
+        ),
+      )
+      .andWhere("application.location.id = :locationId", { locationId })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.Completed,
```

Comment:
Do we need to check here that the application disbursement has not been processed. In that case, from my understanding, the application offering change request cannot be made, right?

Reply:
mm, My undeserving is that it can happen anytime when the application is completed (not archived) irrespective of the disbursement status. If a disbursement already happened, then the offering change assessment will create over wards. correct me if I am wrong @andrepestana-aot  @JasonCTang @dheepak-aot @guru-aot @andrepestana-aot

Reply:
Applications with disbursements sent can be reassessed. There is nothing in this feature stating otherwise.

Reply:
@andrewsignori-aot Do you mean applications with disbursement sent can be requested for offering intensity change?

Reply:
Yes, they can. The disbursements being sent are not a problem or a blocker. Even if both disbursements were sent the application can still be reassessed to give more money to a student or generate an overaward. 
Disbursement sent, so far, does not block anything.

Reply:
Got it. Thank You @andrewsignori-aot